nixos/modules/services/web-apps/rundeck.nix

```diff
@@ -0,0 +1,550 @@
+            fi
+
+            ${lib.getExe cfg.package} \
+              --skipinstall \
+              -b ${cfg.dataDir} \
+              -c ${cfg.configDir} \
+              -p ${cfg.dataDir}/projects
+          ''
+        );
+        WorkingDirectory = cfg.dataDir;
+        RuntimeDirectory = "rundeck";
+        RuntimeDirectoryMode = "0750";
+        UMask = "0027";
+
+        LimitNOFILE = 65536;
+        ReadWritePaths = [
+          cfg.dataDir
+          cfg.configDir
+          cfg.tempDir
+        ];
+        RestartSec = "10s";
+        Restart = "always";
+        TimeoutStartSec = "5min";
+      };
+
+      preStart = ''
+        cat > ${cfg.configDir}/rundeck-config.properties <<EOF
+        ${lib.concatStringsSep "\n" (
+          formatConfig (
+            cfg.settings
+            // {
+              "server.address" = toString cfg.serverAddress;
+              "server.port" = toString cfg.serverPort;
+              "grails.serverURL" =
+                if cfg.serverURL != "" then
+                  cfg.serverURL
+                else
+                  "http${lib.optionalString cfg.ssl.enable "s"}://${cfg.serverHostname}";
+              "logging.config" = "${cfg.configDir}/log4j2.properties";
+              "dataSource.url" =
+                if cfg.database.type == "h2" then
+                  "jdbc:h2:file:${cfg.dataDir}/data/rundeckdb;DB_CLOSE_ON_EXIT=FALSE;LOCK_TIMEOUT=10000;MODE=MYSQL"
+                else if cfg.database.type == "postgresql" then
+                  "jdbc:postgresql://${cfg.database.host}:${toString cfg.database.port}/${cfg.database.name}"
+                else
+                  "jdbc:mysql://${cfg.database.host}:${toString cfg.database.port}/${cfg.database.name}?autoReconnect=true&useSSL=false";
+              "dataSource.driverClassName" =
+                if cfg.database.type == "h2" then
+                  "org.h2.Driver"
+                else if cfg.database.type == "postgresql" then
+                  "org.postgresql.Driver"
+                else
+                  "org.mariadb.jdbc.Driver";
+              "dataSource.username" = cfg.database.username;
+              "dataSource.password" = cfg.database.password;
+            }
+            // cfg.extraConfig
+            // lib.optionalAttrs (cfg.database.type == "h2") {
+              "dataSource.dialect" = "org.hibernate.dialect.H2Dialect";
+            }
+            // lib.optionalAttrs (cfg.database.type == "mysql") {
+              "dataSource.dialect" = "org.hibernate.dialect.MariaDB103Dialect";
+            }
+            // lib.optionalAttrs cfg.ssl.enable {
+              "server.https.port" = "4443";
+              "server.ssl.keyStore" = toString cfg.ssl.keyStore;
+              "server.ssl.keyStorePassword" = cfg.ssl.keyStorePassword;
+              "server.ssl.keyPassword" = cfg.ssl.keyPassword;
+            }
+          )
+        )}
+        EOF
+
+        cat > ${cfg.configDir}/framework.properties <<EOF
```

Comment:
Same here

Reply:
Okay, I'll look into that.

Reply:
tmpfiles would be great for simple non static file.
But in this case im not sure that would be the best approach.